src/java/org/apache/poi/ss/usermodel/DataFormatter.java

```diff
@@ -691,6 +677,57 @@ private String cleanFormatForNumber(String formatStr) {
         return sb.toString();
     }
 
+    private static class InternalDecimalFormatWithScale extends Format {
+
+        private static final Pattern endsWithCommas = Pattern.compile("(,+)$");
+        private BigDecimal divider;
+        private static final BigDecimal ONE_THOUSAND = new BigDecimal(1000);
+        private final DecimalFormat df;
+        private static final String trimTrailingCommas(String s) {
+            return s.replaceAll(",+$", "");
+        }
+
+        public InternalDecimalFormatWithScale(String pattern, DecimalFormatSymbols symbols) {
+            df = new DecimalFormat(trimTrailingCommas(pattern), symbols);
+            setExcelStyleRoundingMode(df);
+            Matcher endsWithCommasMatcher = endsWithCommas.matcher(pattern);
+            if (endsWithCommasMatcher.find()) {
+                String commas = (endsWithCommasMatcher.group(1));
+                BigDecimal temp = BigDecimal.ONE;
```

Comment:
Hi @onealj, 
I'm sorry I didn't understand your comment.
I thought that custom formats were only saved in US locale. With this assumption, the code above simply counts the number of commas _at the end_ of the formatting string and then divide by 1000^n.
Could you please be more clear?
Thank you very much for your help.